spec/features/courses/sites/new_spec.rb

```diff
@@ -94,6 +94,19 @@
     end
   end
 
+  context "Error handling" do
+    let(:course) do
+      c = build(:course, provider: provider, site_ids: [])
+      c.errors.add(:sites, "Invalid")
+      c
```

Comment:
```
build(:course, provider: provider, site_ids: []).then do |course|
  course.errors.add(:sites, "Invalid")
end
```

Reply:
If I'm not mistaken, `then` will return the result of `course.errors.add(:sites, "Invalid")`. `tap` would return the course.

```ruby
build(:course, provider: provider, site_ids: []).tap do |course|
  course.errors.add(:sites, "Invalid")
end
```